### PR TITLE
Add Segment.io integration

### PR DESCRIPTION
**Pull request by silentlight**
- Add Segment.io integration
- Add custom events
- Disable for Figment development

### CODE REVIEW

File: types/types.ts [hunk 1]
```diff
@@ -102,3 +102,8 @@ export type StepType = {
   title: string
   url: string
 }
+
+export enum UserActivity {
+  PROTOCOL_CLICKED = 'PROTOCOL_CLICKED',
```
Comment (guillaumegaluz) [inline types/types.ts:107]: what about `/` loaded?

Comment (silentlight) [reply types/types.ts:107]: This is covered by page event

**Review by guillaumegaluz — APPROVED** [review]
LGTM but I'll let @zurgl give his approval for Bartosz to merge

**Comment by zurgl** [comment]
@silentlight Bartosz, could you manage a scenario with local-storage ?
you can find this on **components/protocols/solana/components/index.tsx** clearStorage fun

**Review by zurgl — APPROVED** [review]

**Comment by guillaumegaluz** [comment]
The goal here is to track the progression through steps so we don’t really
need to handle clearing storage. Let’s merge this so we can have it for
tomorrow’s event

**Review by guillaumegaluz — APPROVED** [review]

**Review by zurgl — APPROVED** [review]

**Comment by guillaumegaluz** [comment]
i merged this @silentlight cause i need to test that this is working in Mixpanel and build a first version of the funnel

**Pull request merged**